Assets/LeapMotion/Core/Scripts/Utils/Utils.cs

```diff
@@ -764,6 +843,118 @@ public struct ChildrenEnumerator : IEnumerator<Transform> {
+    /// B.Then(A.From(B)) == A.
+    /// </summary>
+    public static Quaternion From(this Quaternion thisQuaternion, Quaternion otherQuaternion) {
+      return thisQuaternion * Quaternion.Inverse(otherQuaternion);
+    }
+
+    /// <summary>
+    /// Rotates this quaternion by the other quaternion. This is a rightward syntax for
+    /// Quaternion multiplication, which normally obeys left-multiply ordering.
+    /// </summary>
+    public static Quaternion Then(this Quaternion thisQuaternion, Quaternion otherQuaternion) {
+      return otherQuaternion * thisQuaternion;
+    }
+
+    #endregion
+    
+    #region Float Utils
+
+    /// <summary>
+    /// Additive From syntax for floats. Evaluated as this float plus the additive
+    /// inverse of the other float, usually expressed as thisFloat - otherFloat.
+    /// 
+    /// For less trivial uses of From/Then syntax, refer to their implementations for
+    /// Quaternions and Matrix4x4s.
+    /// </summary>
+    public static float From(this float thisFloat, float otherFloat) {
+      return thisFloat - otherFloat;
+    }
+
+    /// <summary>
+    /// Additive Then syntax for floats. Literally, thisFloat + otherFloat.
+    /// </summary>
+    public static float Then(this float thisFloat, float otherFloat) {
+      return thisFloat + otherFloat;
+    }
+
+    #endregion
+
+    #region Matrix4x4 Utils
+
+    /// <summary>
+    /// A.From(B) produces the matrix that transforms from B to A.
+    /// Combines with Then() to produce readable, predictable results:
+    /// B.Then(A.From(B)) == A.
+    /// 
+    /// Warning: Scale factors of zero will invalidate this behavior.
+    /// </summary>
+    public static Matrix4x4 From(this Matrix4x4 thisMatrix, Matrix4x4 otherMatrix) {
+      return thisMatrix * otherMatrix.inverse;
+    }
+
+    /// <summary>
+    /// Transforms this matrix by the other matrix. This is a rightward syntax for
+    /// matrix multiplication, which normally obeys left-multiply ordering.
+    /// </summary>
+    public static Matrix4x4 Then(this Matrix4x4 thisMatrix, Matrix4x4 otherMatrix) {
+      return otherMatrix * thisMatrix;
+    }
+
+    #endregion
+
+    #region Vector3 Utils
+
+    /// <summary>
+    /// Additive From syntax for Vector3. Literally thisVector - otherVector.
+    /// </summary>
+    public static Vector3 From(this Vector3 thisVector, Vector3 otherVector) {
+      return thisVector - otherVector;
+    }
+
+    /// <summary>
+    /// Additive Then syntax for Vector3. Literally thisVector + otherVector.
+    /// For example: A.Then(B.From(A)) == B.
+    /// </summary>
+    public static Vector3 Then(this Vector3 thisVector, Vector3 otherVector) {
+      return thisVector + otherVector;
+    }
+
+    /// <summary>
+    /// Rightward syntax for applying a Quaternion rotation to this vector; literally
+    /// byQuaternion * thisVector.
+    /// </summary>
+    public static Vector3 Rotate(this Vector3 thisVector, Quaternion byQuaternion) {
```

Comment:
Maybe `RotateBy` instead?  To emphasize that the affected party is the vector instead of the argument?  Also might want to consider a separate  naming convention so not to conflict with things like Vector3.Scale, which does a conceptually similar thing, but _modifies_ the vector instead of returning a new one.

```
Vector3 v;
v.Scale(blah);  //legal, modifies v 
v.Rotate(blah); //legal, but doesn't change v!!!
```

Reply:
OK, will use `RotatedBy`